src/server/rdb_save.cc

```diff
@@ -1251,15 +1251,7 @@ void RdbSaver::Impl::Cancel() {
   if (!shard)
     return;
 
-  auto& snapshot = GetSnapshot(shard);
-  if (snapshot)
-    snapshot->StopChannel();
-
-  dfly::SliceSnapshot::DbRecord rec;
-  while (channel_.Pop(rec)) {
-  }
-
-  snapshot->Join();
+  GetSnapshot(shard)->Cancel();
```

Comment:
This function is called from `RdbSaver::Cancel`. Similarly to how we have `RdbSaver::StartSnapshotInShard` and `StartIncrementalSnapshotInShard`, lets rename it to `RdbSaver::CancelInShard`.
Subsequentially, lets dcheck here that that shard is not-null instead checking &  returning.